docs/tutorial/heaptrack.md

```diff
@@ -0,0 +1,91 @@
+# Heaptrack in Nim Waku
+
+## Background
+Given that RAM is a limited resource, it is crucial to have a good insight on what is going on with the memory used by a particular process.
+
+## Heaptrack
+Heaptrack is a tool that allows to generate memory usage reports.
+It operates in two modes:
+- preload: the tracking is made from the beginning.
+- inject: the tracking starts at a certain time by attaching to a running process.
+
+### Building Heaptrack (tried on Ubuntu)
+- `git clone git@github.com:KDE/heaptrack.git`
+- `mkdir build; cd build`
+- `cmake ..`
+   At this point, make sure the cmake doesn't complain about any missing dependency.
+   Among others, the most tricky deps are obtained by the next commands:
+   - `sudo apt install libkf5i18n-dev`
+   - `sudo apt install libkf5itemmodels-dev`
+   - `sudo apt install libkf5threadweaver-dev`
+   - `sudo apt install libkf5service-dev`
+   - `sudo apt install libkf5completion-dev`
+   - `sudo apt install libkf5itemviews-dev`
+   - `sudo apt install libkf5jobwidgets-dev`
+   - `sudo apt install libkf5solid-dev`
+   - `sudo apt install libkf5coreaddons-dev`
+   - `sudo apt install libkf5auth-dev`
+   - `sudo apt install libkf5codecs-dev`
+   - `sudo apt install libkf5configwidgets-dev`
+   - `sudo apt install libkf5xmlgui-dev`
+   - `sudo apt install libkf5widgetsaddons-dev`
+   - `sudo apt install libqt5gui5`
+   - `sudo apt install libkf5kio-dev`
+   - `sudo apt install libkf5iconthemes-dev`
+- `make`
+- On completion, the `bin/heaptrack_gui` and `bin/heaptrack` binaries will get generated.
+    - heaptrack: needed to generate the report.
+    - heaptrack_gui: needed to analyse the report.
+
+## Heaptrack & Nim Waku
+nwaku supports heaptrack but it needs a special compilation setting.
+
+### Build Nim Waku with heaptrack support
+
+The make command should have the 'NIM_COMMIT' setting as:
+
+`make -j<nproc> NIM_COMMIT=heaptrack_support ...`
+
+This is to force the `nimbus-build-system` to use the Nim compiler that points at the [heaptrack_support](https://github.com/status-im/nim/tree/heaptrack_support) branch.
+
+### Create nwaku memory report with heaptrack
+
+nwaku only works correctly with heaptrack operating in 'preload' mode, i.e. the memory is tracked from the beginning. To achive this, the `heaptrack` binary should be prepended to the usual `wakunode`:
```

Comment:
```suggestion
nwaku only works correctly with heaptrack operating in 'preload' mode, i.e. the memory is tracked from the beginning.
To achieve this, the `heaptrack` binary should be prepended to the usual `wakunode`:
```

With [semantic breaks](https://sembr.org/) each sentence or standalone clause can be in a new line (single newlines are not rendered in markdown, but makes it easier to review docs).